### PR TITLE
add proper smoke test for prototype transforms

### PR DESCRIPTION
**Pull request by pmeier**
Per title. We test two things:

1. Does the structure of the output match the structure of the input?
2. If the transform can handle the input (based on Transform._transformed_types) is the type preserved? If not, does the transform pass the input through?

We test this against a plethora of configurations:

1. CPU / GPU
2. dict, tuple, list inputs
3. `datapoints.Image`, `datapoints.Video`, `PIL.Image.Image`, plain `torch.Tensor` as "core" input

Apart from the "core" input, by default each sample contains more images and videos, bounding boxes in all formats (regular and degenerate ones), detection and segmentation masks, Python builtin types (`int`, `str`), other objects (`pathlib.Path`), a plain tensor, and a `numpy` array.

cc @vfdev-5 @bjuncek

### CODE REVIEW

File: test/test_prototype_transforms.py [hunk 1]
```diff
@@ -66,53 +68,191 @@ def parametrize(transforms_with_inputs):
...
+            (transforms.RandomVerticalFlip(p=1.0), None),
+            (transforms.RandomZoomOut(p=1.0), None),
+            (transforms.Resize([16, 16], antialias=True), None),
+            (transforms.ScaleJitter((16, 16)), None),
+            (transforms.ClampBoundingBoxes(), None),
+            (transforms.ConvertBoundingBoxFormat(datapoints.BoundingBoxFormat.CXCYWH), None),
+            (transforms.ConvertDtype(), None),
+            (transforms.GaussianBlur(kernel_size=3), None),
+            (
+                transforms.LinearTransformation(
```
Comment (pmeier) [inline test/test_prototype_transforms.py:154]: Per comment:

https://github.com/pytorch/vision/blob/707457050620e1f70ab1b187dad81cc36a7f9180/torchvision/prototype/transforms/_misc.py#L82-L83

This of course then fails the type checks added in this PR

File: test/test_prototype_transforms.py [hunk 2]
```diff
@@ -66,53 +68,198 @@ def parametrize(transforms_with_inputs):
...
+        output = transform(input)
+        output_flat, output_spec = tree_flatten(output)
+
+        assert output_spec == input_spec
+
+        for output_item, input_item in zip(output_flat, input_flat):
+            if check_type(input_item, transform._transformed_types):
+                assert type(output_item) is type(input_item)
+            else:
+                assert output_item is input_item
```
Comment (NicolasHug) [inline test/test_prototype_transforms.py:262]: Maybe we can `zip` with the output of `self._needs_transform_list()` and have a stricter check here

File: test/test_prototype_transforms.py (hunk 1 above)
Comment (NicolasHug) [reply test/test_prototype_transforms.py:154]: I'll submit a PR soon to return an Image instead

**Review by NicolasHug — COMMENTED** [review]
Some comments already discussed offline

File: test/test_prototype_transforms.py (hunk 2 above)
Comment (pmeier) [reply test/test_prototype_transforms.py:262]: I have a local commit that I'll push after https://github.com/pytorch/vision/pull/7238#discussion_r1105623206 is done.

**Review by NicolasHug — APPROVED** [review]
LGTM, thanks a lot!

As a follow-up we should try to validate that the degenerate bboxes stay degenerate (or potentially passed-through), although this may involve a mechanism we don't have yet

**Pull request merged**